### PR TITLE
Defensics: instrumentation fix

### PR DESCRIPTION
**Pull request by KKopyscinski**
This patch improves stability of run, allowing to test large sets of test
without crashes of testing sysem. It allows also to implement better
instrumentation methods in the future, as well as tooling using
coap_automation_handler.

Now, instrumentation and tcp server handling defensics packages are running
in separate threads, and do not collide with each other.

### CODE REVIEW

File: defensics/tcp_data_handler.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+from defensics.coap_proxy import CoapProxy
+from defensics.tcp_server import TCPServer
+
+class CoapAutomationHandler(threading.Thread):
+    def __init__(self, proxy: CoapProxy, tcp_server: TCPServer):
+        super().__init__()
+        self.proxy = proxy
+        self.tcp_server = tcp_server
+
+class data_handler(threading.Thread):
```
Comment (michal-narajowski) [inline defensics/tcp_data_handler.py:13]: Class name not following coding style

File: defensics/tcp_data_handler.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import threading
+import logging
+
+from defensics.coap_proxy import CoapProxy
+from defensics.tcp_server import TCPServer
+
+class CoapAutomationHandler(threading.Thread):
```
Comment (michal-narajowski) [inline defensics/tcp_data_handler.py:7]: Why have CoapAutomationHandler and data_handler?

File: defensics/tcp_data_handler.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                except TimeoutError:
+                    logging.debug("Response timeout!")
+                    self.proxy.device_iface.Disconnect()
+                    self.proxy.device_iface.Connect()
+                    logging.debug("Reconnecting")
+                    self.tcp_server.conn.close()
+                    self.tcp_server.conn, self.tcp_server.addr = \
+                        self.tcp_server.sock.accept()
+            else:
+                raise Exception("Proxy not ready")
```
Comment (michal-narajowski) [inline defensics/tcp_data_handler.py:37]: Missing new line at the end.

File: defensics/coap_automation_handler.py [hunk 4]
```diff
@@ -158,3 +116,17 @@ def get_status(self):
 
     def stop(self):
         self.stop_event.set()
+
+    def parallel(func):
+        def parallel_func(*args, **kw):
+            p = Process(target=func, args=args, kwargs=kw)
```
Comment (michal-narajowski) [inline defensics/coap_automation_handler.py:122]: We should have a way to gracefully stop the process on teardown.

File: defensics/coap_automation_handler.py [hunk 5]
```diff
@@ -85,53 +61,35 @@ def process(self, job):
...
-        #             logging.error('Exception: {}'.format(str(e)))
-        #             self.status.errors.append(str(e))
-        #             self.status.verdict = 'fail'
-        #         finally:
-        #             if self.processing_lock.locked():
-        #                 self.processing_lock.release()
-        # return
+                    self.process(item)
+                except socket.timeout as e:
+                    logging.error('Exception: BTP Timeout')
```
Comment (michal-narajowski) [inline defensics/coap_automation_handler.py:74]: We don't use BTP here, right?

Comment (KKopyscinski) [reply defensics/coap_automation_handler.py:74]: Right :)

File: defensics/tcp_data_handler.py (hunk 2 above)
Comment (KKopyscinski) [reply defensics/tcp_data_handler.py:7]: Oh, sure, missed leftover

**Pull request closed without merging**